packages/tiled/lib/src/common/property.dart

```diff
@@ -33,14 +33,21 @@ class Property {
 }
 
 extension PropertiesParser on Parser {
-  List<Property> getProperties() {
-    return formatSpecificParsing(
+  Map<String, Property> getProperties() {
+    final properties = formatSpecificParsing(
       (json) => json.getChildrenAs('properties', Property.parse),
       (xml) =>
           xml
               .getSingleChildOrNull('properties')
               ?.getChildrenAs('property', Property.parse) ??
           [],
     );
+
+    return properties.groupFoldBy((prop) => prop.name, (previous, element) {
+      if (previous != null) {
+        throw ArgumentError("Can't have two properties with the same name.");
```

Comment:
This seems like a [state error](https://api.flutter.dev/flutter/dart-core/StateError-class.html)